File: KutuphaneAPI/Entities/Models/Book.cs
namespace Entities.Models
{
    public class Book
    {
        public int Id { get; set; }
        public String? ISBN { get; set; }
        public String? Title { get; set; }
        public int TotalCopies { get; set; }
        public int AvailableCopies { get; set; }
        public String? Location { get; set; }
        public DateTime PublishedDate { get; set; }
        public String? Summary { get; set; }
        public double AverageRating { get; set; } = 0;
        public ICollection<BookImage>? Images { get; set; }
        public ICollection<Tag>? Tags { get; set; }
        public ICollection<Author>? Authors { get; set; }
        public ICollection<Category>? Categories { get; set; }
        public ICollection<Review>? Reviews { get; set; }
        public ICollection<CartLine>? CartLines { get; set; }
        public ICollection<LoanLine>? LoanLines { get; set; }
    }
}

File: KutuphaneWeb/library-automation-web/src/types/book.ts
import type Author from "./author";
import type BookImage from "./bookImage";
import type Category from "./category";
import type Tag from "./tag";
import type UserReview from "./userReview";

export default interface Book {
    id: number | null;
    isbn: string | null;
    title: string | null;
    totalCopies: number | null;
    availableCopies: number | null;
    location: string | null;
    publishedDate: Date | null;
    summary: string | null;
    averageRating: number | null;
    images: BookImage[] | null;
    tags: Tag[] | null;
    authors: Author[] | null;
    categories: Category[] | null;
    reviews?: UserReview[] | null;
}